python/lsst/ts/salobj/sal_info.py

```diff
@@ -792,8 +788,14 @@ def _blocking_read(
             data_list = [
                 self._sample_to_data(sd, si) for sd, si in data_list if si.valid_data
             ]
-            return (reader, data_list)
-        return (None, [])
+            if not data_list:
+                ret.append((reader, data_list))
+
+        if not self.isopen:
+            # shutting down; clean everything up
+            return []
```

Comment:
The main reason for this test is to avoid trying to read DDS data after the SalInfo is closed, for fear that the DDS system itself may be shut down, which results in confusing messages.

The safest place to put it is in the loop over conditions -- i.e. before reading DDS data from each reader. I moved it there in the commit I just pushed.

It could also go before the loop, which reduces the # of tests if there is more than one condition, but adds a small interval over which the system may be closed but the loop tries to read data. Or we could skip it, but I'd prefer to test it at least once before trying to read any data from a DDS reader.

Reply:
I am Ok with that - unaware of Reader interaction, I placed test after the loop. Fine with whenever it ends up.